### ISSUE TITLE
Documentation for tf.train.init_from_checkpoint doesn't say what it does when

### ISSUE THREAD

**Issue by girving** [tensorflow/tensorflow#22452]
Base on the code, [`tf.train.init_from_checkpoint`](https://www.tensorflow.org/api_docs/python/tf/train/init_from_checkpoint) seems to overwrite the initializers of variables to load from the given checkpoint.  The actual initialization will then occur when `tf.global_variables_initializer()` or similar is executed.

However, the documentation makes no mention of this.  It's written as if the initialization happens *now*.  I suppose it does happen now for eager mode, but for Graph mode the initialization is delayed.

This is particularly confusing in a TPU context where code is executing in all sorts of places, so it took me a while reading through the underlying source code before I understood what was going on.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by MarkDaoust** [comment]
@vihanjain: could you take a look?

**Comment by girving** [comment]
Thank you!
